pkg/runtime/pubsub/bulksubscribe_events.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2022 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pubsub
+
+import (
+	"github.com/google/uuid"
+
+	contribPubsub "github.com/dapr/components-contrib/pubsub"
+)
+
+const Metadata = "metadata"
+
+type BulkSubscribeMessageItem struct {
+	EntryID     string            `json:"entryID"`
+	Event       interface{}       `json:"event"`
+	Metadata    map[string]string `json:"metadata"`
+	ContentType string            `json:"contentType,omitempty"`
+}
+
+// type BulkSubscribeRawPayloadMessageItem struct {
+// 	EntryID     string            `json:"entryID"`
+// 	Event       string            `json:"event"`
+// 	ContentType string            `json:"contentType,omitempty"`
+// 	Metadata    map[string]string `json:"metadata"`
+// }
+
+type BulkSubscribeEnvelope struct {
+	ID        string
+	Entries   []BulkSubscribeMessageItem
+	Metadata  map[string]string
+	Topic     string
+	Pubsub    string
+	EventType string
+}
+
+func NewBulkSubscribeEnvelope(req *BulkSubscribeEnvelope) map[string]interface{} {
+	// defaults
+	id := req.ID
+	if id == "" {
+		id = uuid.New().String()
```

Comment:
If there isn't enough entropy in the kernel, this will panic. Do we use this emthod in other hot paths? Otherwise, would be better to use the method that returns an error rather than panics in case of lack of entropy.

Reply:
Agreed, good suggestion. I have used uuid.NewRandom() instead now.